File: ASG1sub1.pl
#! /usr/bin/perl

use strict;
use warnings;
use Encode;

my $filename = "./NazrulYusni-perl.txt";

my $name = "Nazrul Yusni bin Hamzah\n";
my $address = "Seksyen U16 Shah Alam\n";

open(my $FH, '>', $filename) or die "Could not open file '$filename' $!";
print $FH $name;
print $FH $address;
close $FH;

File: ASG1sub1.java
import static java.nio.file.StandardOpenOption.*;
import java.nio.file.*;
import java.io.*;


public class ASG1sub1 {

  public static void main(String[] args) {

    // Convert the string to a
    // byte array.
    String name = "Nazrul Yusni bin Hamzah\n";
    String address = "No4 Jalan Elektron U16/52D Shah Alam\n";
    byte data1[] = name.getBytes();
    byte data2[] = address.getBytes();
    Path p = Paths.get("./NazrulYusni.txt");

    try (OutputStream out = new BufferedOutputStream(Files.newOutputStream(p, CREATE))) {
       out.write(data1, 0, data1.length);
       out.write(data2, 0, data2.length);
    } 
    catch (IOException x) {
      System.err.println(x);
    }
  }
}

